src/main/org/firebirdsql/management/FBStreamingBackupManager.java

```diff
@@ -0,0 +1,321 @@
+/*
+ * Firebird Open Source JavaEE Connector - JDBC Driver
+ *
+ * Distributable under LGPL license.
+ * You may obtain a copy of the License at http://www.gnu.org/copyleft/lgpl.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * LGPL License for more details.
+ *
+ * This file was created by members of the firebird development team.
+ * All individual contributions remain the Copyright (C) of those
+ * individuals.  Contributors to this file are either listed here or
+ * can be obtained from a source control history command.
+ *
+ * All rights reserved.
+ */
+package org.firebirdsql.management;
+
+import org.firebirdsql.gds.ServiceRequestBuffer;
+import org.firebirdsql.gds.ServiceParameterBuffer;
+import org.firebirdsql.gds.impl.GDSType;
+import org.firebirdsql.gds.ng.FbService;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Arrays;
+import java.io.OutputStream;
+import java.io.InputStream;
+import java.io.BufferedInputStream;
+
+import static org.firebirdsql.gds.ISCConstants.*;
+import static org.firebirdsql.gds.VaxEncoding.iscVaxInteger2;
+import static org.firebirdsql.gds.VaxEncoding.iscVaxInteger;
+
+/**
+ * Implements the streaming version of the backup and restore functionality of
+ * Firebird Services API.
+ *
+ * @author <a href="mailto:rrokytskyy@users.sourceforge.net">Roman Rokytskyy</a>
+ * @author <a href="mailto:mrotteveel@users.sourceforge.net">Mark Rotteveel</a>
+ */
+public class FBStreamingBackupManager extends FBBackupManagerBase implements BackupManager {
+
+    private OutputStream backupOutputStream = null;
+    private BufferedInputStream restoreInputStream = null;
+
+    private int backupBufferSize = BUFFER_SIZE * 30; // 30K
+    private final int MAX_RESTORE_CHUNK = Short.MAX_VALUE;
```

Comment:
I did some testing and with SPB version 2 (that Jaybird currently uses for all Firebird versions, even Firebird 3), the maximum seems to be 65534 (or `2^16 - 1 - 2`, where the 2 is probably the 2 bytes for length). I haven't tested this against all Firebird versions though.

I think we should leave this as is right now, but when I fix Jaybird to use the latest SPB for the Firebird version, I think we can use the maximum buffer size supported by the Firebird version.  So with the version 3 SPB for Firebird 3 that would be an even bigger buffer.